doc/NAC/PnacSonicHLD.md

```diff
@@ -0,0 +1,374 @@
+***Note***: The standard supports port-based authentication and MAC based authentication. This version of the software supports port based authentication and will be extended for MAC based authentication in future.
+# Scope
+This document describes the high level design details about Port Based Authentication.
+# Definitions/Abbreviations
+###### Table 1: Abbreviations 
+#
+| Abbreviation             | Full form                              |
+|---------------------------|----------------------------------|
+| AAA                  | Authentication, Authorization, and Accounting |
+| EAPoL               | Authentication Protocol (EAP) over LAN |
+| MAC                 | Media Access Control   |
+| NAC                             | Network Access Control     |
+| PAE                              | Port Authentication Entity |
+| PNAC                      | Port Based Network Access Control |
+| VLAN                     | Virtual Local Area Network |
+
+# 1 Requirements Overview
+Port-based network access control allows a network administrator to restrict the use of IEEE 802 LAN service access points (ports) to secure communication between authenticated and authorised devices. The current design is restricted to local switch based authentication and authorisation method. 
+## 1.1 Functional Requirements
+The functional requirements for PNAC are:
+1. The software allows a user to enable/disable the NAC functionality globally in the system.
+User can configure NAC per port only if NAC is globally enabled.
+2. The software allows a user to enable the PNAC functionality on any specific physical port in the system.
+3. The software allows a user to disable the PNAC functionality on any specific physical port in the system.
+4. The Software should perform periodic authentication of the supplicant.
+5. The software includes a CLI for a user to configure and monitor the feature.
+## 1.2 Memory, CPU Consumptions Requirements
+1. The software memory usage must be proportional to the number of physical ports.
+2. The software limits for the RAM overhead and CPU overhead due to usage of a separate container shall be within allowed limits of SONiC container.
+
+## 1.3 Configuration and Management Requirements
+Configuration of the PNAC feature can be done via:
+- JSON config input
+- CLI
+# 2 Functionality
+The feature enables a switch to authenticate other devices that require access to the network through the switch. Such end devices could be cameras, IP-phones, servers, user workstations, WiFi access points, etc. The administrator is expected to have the following choices / controls.
+- Enable / disable authentication globally
+- Enable / disable authentication on each port
+- Perform local authentication with secret in plain text 
```

Comment:
Remote authentication always preferred over local authentication as it is helps with Centralized Provisioning of clients. Also it is scalable as more and more Client profile attributes are supported by Remote AAA servers.

Reply:
Please provide CLI command details of how the local user database is configured.

Reply:
Current release supports static user configuration by updating "hostapd configuration file". There are no CLI supported but in future release AAA will be part of the roadmap feature.